### PR TITLE
Add support for rust packages

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: TG1999 <tushar.goel.dav@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
Hi, I have added support for crates and also added tests for it, Kindly review it :)

**Comment by TG1999** [comment]
Solves issue #23

File: src/packageurl/contrib/url2purl.py [hunk 1]
```diff
@@ -263,3 +263,16 @@ def build_nuget_purl(uri):
 @purl_router.route(sourceforge_pattern)
 def build_sourceforge_purl(uri):
     return purl_from_pattern('sourceforge', sourceforge_pattern, uri)
+
+
+crates_pattern= (
+    r"^https?://crates.io/api/v1/crates/"
+    r"(?P<name>.+)/(?P<version>.+)"
+    r"(\/download)$"
+)
```
Comment (majurg) [inline src/packageurl/contrib/url2purl.py]: I am not sure we want to rely on regex for constructing/deconstructing purls

Comment (TG1999) [reply src/packageurl/contrib/url2purl.py]: Hi, regex has been used above for Ruby, nugget , pypi and sourceforge so I think it is good to go

Comment (majurg) [reply src/packageurl/contrib/url2purl.py]: fair enough.

**Comment by TG1999** [comment]
Thanks @majurg

**Comment by pombredanne** [comment]
See https://github.com/package-url/packageurl-python/issues/23#issuecomment-603910565 ... we specified cargo in the spec... but I would like to get Rust folks input too.

**Comment by TG1999** [comment]
Yeah sure :), waiting for their input

**Comment by TG1999** [comment]
Hi @pombredanne I have gone through the spec, should we wait for input or should I change it to cargo.

**Comment by TG1999** [comment]
@pombredanne updated

**Pull request merged**